### PR TITLE
Implement script for quest 11664

### PR DESCRIPTION
**Pull request by Grz3s**

### CODE REVIEW

File: src/scriptdev2/scripts/northrend/borean_tundra.cpp [hunk 1]
```diff
@@ -1195,69 +1196,195 @@ CreatureAI* GetAI_npc_seaforium_depth_charge(Creature* pCreature)
...
+	{
+		if (!HasEscortState(STATE_ESCORT_ESCORTING))
+			m_creature->SetStandState(UNIT_STAND_STATE_SIT);
+	}
+
+	void WaypointReached(uint32 uiPointId) override
+	{
+		switch (uiPointId)
+		{
+			case 1:
```
Comment (Fabi) [inline src/scriptdev2/scripts/northrend/borean_tundra.cpp]: spaces ftw :D

File: src/scriptdev2/scripts/northrend/borean_tundra.cpp [hunk 2]
```diff
@@ -1195,69 +1196,195 @@ CreatureAI* GetAI_npc_seaforium_depth_charge(Creature* pCreature)
...
+	pNewScript->Name = "npc_seaforium_depth_charge";
+	pNewScript->GetAI = &GetAI_npc_seaforium_depth_charge;
+	pNewScript->RegisterSelf();
+
+	pNewScript = new Script;
+	pNewScript->Name = "npc_mootoo_the_younger";
+	pNewScript->GetAI = &GetAI_npc_mootoo_the_youngerAI;
+	pNewScript->pQuestAcceptNPC = &QuestAccept_npc_mootoo_the_younger;
+	pNewScript->RegisterSelf();
+}
```
Comment (Fabi) [inline src/scriptdev2/scripts/northrend/borean_tundra.cpp]: I think we love new lines?

Comment (Grz3s) [inline src/scriptdev2/scripts/northrend/borean_tundra.cpp]: :japanese_goblin: Leave a comment

File: src/scriptdev2/scripts/northrend/borean_tundra.cpp [hunk 3]
```diff
@@ -1195,6 +1196,126 @@ CreatureAI* GetAI_npc_seaforium_depth_charge(Creature* pCreature)
...
+                break;
+            case 24:
+                DoScriptText(SAY_9_MOOTOO_Y, m_creature);
+                if (Player* pPlayer = GetPlayerForEscort())
+                if (Creature* pFather = GetClosestCreatureWithEntry(m_creature, NPC_ELDER_MOOTOO, 30.0f))
+                    DoScriptText(SAY_2_ELDER_MOOTOO, pFather, pPlayer);
+                break;
+            case 25:
+                if (Player* pPlayer = GetPlayerForEscort())
+                if (Creature* pFather = GetClosestCreatureWithEntry(m_creature, NPC_ELDER_MOOTOO, 30.0f))
```
Comment (xfurry) [inline src/scriptdev2/scripts/northrend/borean_tundra.cpp]: Add some extra spaces here, or surround by { .... }. It makes the code look better and cleaner.
Same for above.

Everything else looks good.

File: src/scriptdev2/scripts/northrend/borean_tundra.cpp [hunk 4]
```diff
@@ -1195,6 +1196,126 @@ CreatureAI* GetAI_npc_seaforium_depth_charge(Creature* pCreature)
...
+        if (eventType == AI_EVENT_START_ESCORT && pInvoker->GetTypeId() == TYPEID_PLAYER)
+            Start(false, (Player*)pInvoker, GetQuestTemplateStore(uiMiscValue));
+    }
+};
+
+bool QuestAccept_npc_mootoo_the_younger(Player* pPlayer, Creature* pCreature, const Quest* pQuest)
+{
+    if (pQuest->GetQuestId() == QUEST_ESCAPING_THE_MIST)
+        pCreature->AI()->SendAIEvent(AI_EVENT_START_ESCORT, pPlayer, pCreature, pQuest->GetQuestId());
+    return true;
```
Comment (xfurry) [inline src/scriptdev2/scripts/northrend/borean_tundra.cpp]: Include ```return true;``` in the if statement and surround by { }.
And add a ``` return false;``` statement at the end.

Reason for this, is the following:
If the NPC has more quests, if you always return true in SD2, you won't be able to use dbscripts_on_quest_start / end for any other quest.

Comment (Grz3s) [inline src/scriptdev2/scripts/northrend/borean_tundra.cpp]: @xfurry 
```code
bool QuestAccept_npc_mootoo_the_younger(Player* pPlayer, Creature* pCreature, const Quest* pQuest)
{
	if (pQuest->GetQuestId() == QUEST_ESCAPING_THE_MIST)
	{
		pCreature->AI()->SendAIEvent(AI_EVENT_START_ESCORT, pPlayer, pCreature, pQuest->GetQuestId());
		return true;
	}

	return false;
}
```
correct?

File: src/scriptdev2/scripts/northrend/borean_tundra.cpp (hunk 3 above)
Comment (Grz3s) [inline src/scriptdev2/scripts/northrend/borean_tundra.cpp]: @xfurry 
```code
            case 24:
                DoScriptText(SAY_9_MOOTOO_Y, m_creature);
                if (Player* pPlayer = GetPlayerForEscort())
					if (Creature* pFather = GetClosestCreatureWithEntry(m_creature, NPC_ELDER_MOOTOO, 30.0f))
						DoScriptText(SAY_2_ELDER_MOOTOO, pFather, pPlayer);
                break;
            case 25:
                if (Player* pPlayer = GetPlayerForEscort())
					if (Creature* pFather = GetClosestCreatureWithEntry(m_creature, NPC_ELDER_MOOTOO, 30.0f))
						pFather->HandleEmote(EMOTE_ONESHOT_BOW);
                SetEscortPaused(true);
                m_creature->ForcedDespawn(10000);
                break;

File: src/scriptdev2/scripts/northrend/borean_tundra.cpp (hunk 2 above)
Comment (Ulduar) [inline src/scriptdev2/scripts/northrend/borean_tundra.cpp]: http://www.wowhead.com/item=90087/lobstmourne
Hungers...

**Comment by xfurry** [comment]
I think that someone can merge this PR. Unfortunately I'm not available to do it myself at the moment (until end of Feb).
Poke @cyberium or @Warlockbugs

**Comment by cyberium** [comment]
Ok iam on it

**Comment by cyberium** [comment]
Merged thanks

**Pull request closed without merging**